### DISCUSSION TITLE
Using a token to authenticate server client ( Clerk + Supabase )

### DISCUSSION THREAD

**Discussion by ojkaas** [supabase/supabase#20271] [Questions]
I am using clerk in combination with supabase, i have been moving some of my logic into server side rendering / actions, but am struggling how to get supabase serverClient to work with the access token generated by clerk.

Before i did this to get the proper access for a user:
```

const supabaseClient = async (supabaseAccessToken) => {
    const supabase = createClient(process.env.NEXT_PUBLIC_SUPABASE_URL, process.env.NEXT_PUBLIC_SUPABASE_KEY, {
      global: { headers: { Authorization: `Bearer ${supabaseAccessToken}` } },
    });
    // set Supabase JWT on the client object,
    // so it is sent up with all Supabase requests
    return supabase;
  }
  
```

This makes it that my RLS is working properly based on the user that is authenticated. But I have no idea how to do this with the new package and cannot find any info on it, as everywhere it is mentioned that is needs cookies to authenticate, any idea how to setup a server client with an token?

```
export async function getServerClient(token?: string | null): Promise<SupabaseClient<Database>> {
  const serverClient = createServerClient<Database>(process.env.NEXT_PUBLIC_SUPABASE_URL!, process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!, {
    cookies: {
      get(name: string) {
        if (token) //do something here so that the user is authenticated and i can use my RSL
        return new RequestCookies(headers()) as any
      },
    },
  })

 return serverClient
```

  Thanks!

Olaf

**Comment by ikupenov** [comment]
Have you checked https://clerk.com/docs/backend-requests/making/jwt-templates?
